### PR TITLE
Check memory storage before displaying home screen, cache fixtureLoadingService

### PR DESCRIPTION
**Pull request by karlwestin**
Suggested for Issue #51

### CODE REVIEW

File: app/scripts/services/fixture-loader-service.js [hunk 1]
```diff
@@ -169,18 +169,25 @@ angular.module('lmisChromeApp')
...
         promises[dbName] = storageService.get(dbName);
       }
-      return $q.all(promises)
-        .then(indexByID)
-        .then(function(results) {
-          loadDatabasesIntoMemoryStorage(results);
-          return results;
-        });
+
+      var key = JSON.stringify(dbNames);
```
Comment (tlvince) [inline app/scripts/services/fixture-loader-service.js]: Seems a little confusing to use a stringified array as a key:

```
Object {["uom","uom_category","product_types","product_category","product_presentation","ccei","mode_of_administration","product_formulations","product_profiles"]: Object}
```

**Comment by tlvince** [comment]
This does the trick. LGTM.

**Comment by tlvince** [comment]
NB, good idea with the cache. The amount of times this get hit is telling. Can we move the fixture loader call higher up in the state chain (so as not to be called each time the home page is navigated to) perhaps? (Separate task).

File: app/scripts/services/fixture-loader-service.js (hunk 1 above)
Comment (karlwestin) [inline app/scripts/services/fixture-loader-service.js]: im open to suggestions, just seemed like the easiest way to go to me :)

Comment (tlvince) [inline app/scripts/services/fixture-loader-service.js]: Yeah, maybe just a comment'll do.

**Comment by tlvince** [comment]
:stars:

**Pull request merged**